### PR TITLE
Feature Updated.

### PR DESCRIPTION
**Pull request by raunak-ra**

### CODE REVIEW

File: Logger/ApiLogger.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Logger
+{
+    public class ApiLog : LogBase
+    {
```
Comment (Tavisca-smohite) [inline Logger/ApiLogger.cs]: please all the fields needed in apilog

File: Logger/LogBase.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+using System.Text;
+
+namespace Logger
+{
+    public abstract class LogBase
+    {
+        public string SessionId { get; set; }
+        public abstract string Type { get; }
+        public string ApplicationName { get; set; }
+        public string Method { get; set; }
```
Comment (Tavisca-smohite) [inline Logger/LogBase.cs]: please add support for extra fields addition in base class

File: Logger/TraceLogger.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+using System.Text;
+
+namespace Logger
+{
+    public class TraceLogger : LogBase
+    {
+        public override string Type { get => "trace"; }
+        public string Message { get; set; }
+        public string Status { get; set; }
+        public string TimeElapsed { get; set; }
```
Comment (Tavisca-smohite) [inline Logger/TraceLogger.cs]: should always mention whether it's time taken in milliseconds or seconds

File: Logger/TxsLogger.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+
+        public async Task WriteLogAsync(LogBase logger)
+        {
+            var data = logger.GetData();
+
+            if (logger is ErrorLogger)
+            {
+                Log.Error("{@data}", data);
```
Comment (Tavisca-smohite) [inline Logger/TxsLogger.cs]: should log against session id

**Review by Tavisca-smohite — CHANGES_REQUESTED** [review]
1. PR should be raised against develop and not master
2. please add test cases

File: Logger/ApiLogger.cs (hunk 1 above)
Comment (Tavisca-smohite) [reply Logger/ApiLogger.cs]: what about this ?

File: Logger/TraceLogger.cs [hunk 5]
```diff
@@ -9,15 +9,15 @@ public class TraceLogger : LogBase
...
+        public string TimeElapsedInMilliSeconds { get; set; }
 
         public override List<KeyValuePair<string, object>> GetData()
         {
             var data = base.GetData();
             data.Add(new KeyValuePair<string, object>("type", this.Type));
             data.Add(new KeyValuePair<string, object>("message", this.Message));
             data.Add(new KeyValuePair<string, object>("status", this.Status));
-            data.Add(new KeyValuePair<string, object>("time_elapsed", this.TimeElapsed));
+            data.Add(new KeyValuePair<string, object>("time_elapsed", this.TimeElapsedInMilliSeconds));
```
Comment (Tavisca-smohite) [inline Logger/TraceLogger.cs]: the name of the field logged should also be updated

**Review by Tavisca-smohite — COMMENTED** [review]
When you call a class TraceLogger (for example) it implies, the class owns the responsibility of logging thr trace ?
is it logging the trace ?
 if not then name is invalid. Please update the names of the classes as per their responsibilites

File: Logger.Tests/LoggerFixture.cs [hunk 6]
```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Logger.Tests
+{
+    public class LoggerFixture
+    {
+        private readonly ITxsLogger l = new TxsLogger();
```
Comment (ndhande-Tavisca) [inline Logger.Tests/LoggerFixture.cs]: Please give valid name to ITxsLogger object instead of l

File: Logger.Tests/LoggerFixture.cs [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+        public string PropA { get; set; }
+        public string PropB { get; set; }
```
Comment (ndhande-Tavisca) [inline Logger.Tests/LoggerFixture.cs]: these property not understandable please give valid name. please discuss

File: Logger.Tests/LoggerFixture.cs [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+        [Fact]
+        public void Testing_for_apilogger()
```
Comment (ndhande-Tavisca) [inline Logger.Tests/LoggerFixture.cs]: Test name should be like this  APILog_Should_Be_Write_Log_In_File

File: Logger.Tests/LoggerFixture.cs (hunk 6 above)
Comment (raunak-ra) [reply Logger.Tests/LoggerFixture.cs]: done

File: Logger.Tests/LoggerFixture.cs (hunk 8 above)
Comment (raunak-ra) [reply Logger.Tests/LoggerFixture.cs]: done

**Review by ndhande-Tavisca — APPROVED** [review]

**Pull request merged**